openedx_tagging/core/tagging/api.py

```diff
@@ -129,13 +129,7 @@ def resync_object_tags(object_tags: QuerySet | None = None) -> int:
     if not object_tags:
         object_tags = ObjectTag.objects.select_related("tag", "taxonomy")
 
-    num_changed = 0
-    for object_tag in object_tags:
-        changed = object_tag.resync()
-        if changed:
-            object_tag.save()
-            num_changed += 1
-    return num_changed
+    return ObjectTag.resync_object_tags(object_tags)
```

Comment:
What's the advantage of making this a static method on `ObjectTag` rather than leaving it here in the API? I kind of prefer having it here since it forces usage of `api`, and we don't want others directly calling methods on `ObjectTag`. Though perhaps we can use the [isolated apps linter](https://github.com/openedx/edx-platform/blob/febcccc14767378492a7a897b66023dafafda619/setup.cfg#L163-L173) to enforce that in the future as well.

I don't have a strong opinion on this, so no need to change it if you think this makes more sense.

Reply:
I think @yusuf-musleh  did this so he could call `resync` from the `Taxonomy.delete_tags` and `Taxonomy.update_tags` model methods without creating a circular dependency.

But instead, we could leave `resync_object_tags` as-is, and call it from the api `update_tag_in_taxonomy` and `delete_tags_from_taxonomy` methods instead.

What do you think, @yusuf-musleh ?

Reply:
Also, `resync_object_tags` could take a `taxonomy` as its first argument instead of the ObjectTags queryset, since we're currently only resyncing object tags for single taxonomy.

Reply:
I initially moved the logic of `resync_object_tags` to be a static method in the model to avoid duplicate code and to avoid calling api functions from inside the model's methods (like `update_tags`). However I like @pomegranited's suggestion,  makes more sense, I took out the resync logic and placed in in the api and brought back the `resync_object_tags` to the api, removing the static method: https://github.com/openedx/openedx-learning/commit/bdaffeb0d0e718ce56321be6db9a64b83e64bc47

PS: I actually forgot to do the resync of object tags in the delete method, I just added it to the `delete_tags_from_taxonomy`. Thanks for the reminder!